### ISSUE TITLE
VideoCapture Error

### ISSUE THREAD

**Issue by jonhare** [openimaj/openimaj#10]
OP-10: 

I ran this simple application:

/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package tools;

import java.util.List;
import org.openimaj.video.capture.Device;
import org.openimaj.video.capture.VideoCapture;

/**
 *
 * @author JGodino
 */
public class mostrarDispositivos {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {

        List<Device> devices;
        devices = VideoCapture.getVideoDevices();

        for (final Device device : devices) {

            System.out.println(device.toString());

        }

    }

}

In windows I'm receiving the following message:

C:\novovia>java -cp RadarNovovia01.jar tools.mostrarDispositivos
Not enough args for boolean OpenIMAJGrabber.startSession(int, int, double)

***** VIDEOINPUT LIBRARY - 0.1995 - TFW07 *****

VIDEOINPUT SPY MODE!

SETUP: Looking For Capture Devices
SETUP: 0) Integrated Webcam
SETUP: 1) Google Camera Adapter 0
SETUP: 2) Google Camera Adapter 1
SETUP: 3 Device(s) found

Device[0]="Integrated Webcam"
Device[1]="Google Camera Adapter 0"
Device[2]="Google Camera Adapter 1"

In Linux (Ubuntu 12) the same applications:

Not enough args for boolean OpenIMAJGrabber.startSession(int, int)
Not enough args for boolean OpenIMAJGrabber.startSession(int, int)
Device[/dev/video0]="WebCam"

Another issue is the following:

In other application I have:

List<Device> devices;
devices = VideoCapture.getVideoDevices();
Device device = devices.get(0); // WebCam.. It's a Integer
oVideoCapture = new VideoCapture( 320, 240, 30, device );

The result in linux is 

Not enough args for boolean OpenIMAJGrabber.startSession(int, int)
Not enough args for boolean OpenIMAJGrabber.startSession(int, int)
SEVERE: UIApplicacionPpal.MonitorDeCamarasApp$ActivacionDeCamara@f47bf5 failed: java.lang.UnsatisfiedLinkError: org.openimaj.video.capture.OpenIMAJGrabber.startSession(IIDLorg/bridj/Pointer;)Z
java.lang.UnsatisfiedLinkError: org.openimaj.video.capture.OpenIMAJGrabber.startSession(IIDLorg/bridj/Pointer;)Z
	at org.openimaj.video.capture.OpenIMAJGrabber.startSession(Native Method)
	at org.openimaj.video.capture.VideoCapture.startSession(VideoCapture.java:135)
	at org.openimaj.video.capture.VideoCapture.<init>(VideoCapture.java:99)
	at UIApplicacionPpal.MonitorDeCamarasApp$ActivacionDeCamara.doInBackground(MonitorDeCamarasApp.java:261)

I have a question:

How I can use this code in linux:

Device device = devices.get(0); // WebCam.. It's a Integer

I ask this because the id of the devices is: Device[/dev/video0]="WebCam"

**Comment by jonhare** [comment]
Original reporter: @

**Comment by jonhare** [comment]
:  

Related to this issue: 

*Not enough args for boolean OpenIMAJGrabber*

When I run this code:

{code:title=my code|borderStyle=solid}
List<Device> devices;
devices = VideoCapture.getVideoDevices();
{code}

I'm receiving the following message both windows and linux:

{code:title=Message|borderStyle=solid}
Not enough args for boolean OpenIMAJGrabber.startSession(int, int, double)
{code}
I did a research and I found this:




{code:title=VideoCapture.java|borderStyle=solid}
	public static List<Device> getVideoDevices() {
		OpenIMAJGrabber grabber = new OpenIMAJGrabber();
		DeviceList list = grabber.getVideoDevices().get();

		return list.asArrayList();
	}

{code}


{code:title=OpenIMAJGrabber.java|borderStyle=solid}
	public native Pointer<DeviceList> getVideoDevices();

	public native boolean startSession(int width, int height, double reqFPS);
	public native boolean startSession(int width, int height, double reqFPS, Pointer<Device> devptr);
{code}


{code:title=OpenIMAJGrabber.cpp Windows|borderStyle=solid}
bool OpenIMAJGrabber::startSession(int width, int height,double rate)
{
	if(this->getVideoDevices()->getNumDevices() <= ((videoData*)this->data)->device)
		return false;
	return this->startSession(
		width,height,rate,
		this->getVideoDevices()->getDevice(((videoData*)this->data)->device)
	);
}
{code} 

I think the message is generated when the Dll is loaded. 
Is is possible that I have a  previous DLL?

Thanks

Javier

**Comment by jonhare** [comment]
:  Related to this code:


{code:title=My code for linux|borderStyle=solid}
List<Device> devices;
devices = VideoCapture.getVideoDevices();
*Device device = devices.get(0);*
oVideoCapture = new VideoCapture( 320, 240, 30, device );
{code}

The result in linux is:

SEVERE: UIApplicacionPpal.MonitorDeCamarasApp$ActivacionDeCamara@f47bf5 failed: java.lang.UnsatisfiedLinkError: org.openimaj.video.capture.OpenIMAJGrabber.startSession(IIDLorg/bridj/Pointer;)Z
java.lang.UnsatisfiedLinkError: org.openimaj.video.capture.OpenIMAJGrabber.startSession(IIDLorg/bridj/Pointer;)Z

	
Javier

**Comment by jonhare** [comment]
jonhare:  The "Not enough args for boolean OpenIMAJGrabber.startSession(int, int, double)" is a random println in Bridj, the library that we use to link to the native code. I'll file an issue to get that fixed in the next Bridj release.

Regarding the second issue, are you using OpenIMAJ 1.0.5? I think when we made the release we inadvertantly had an old version of the linux shared library included. I thought we'd fixed it by re-deploying a fixed version of the jar, but perhaps not. I'll check tomorrow morning when I've got access to my linux box at work.
